### PR TITLE
Gravatar: Customize the "Welcome back!" view of the passwordless login flow

### PR DESCRIPTION
**Pull request by wellyshen**
Related to 105814-Automattic/gravatar

## Proposed Changes

* To customize the "Welcome back!" view for Gravatar-powered clients

## Testing Instructions

* Sandbox `public-api.wordpress.com`, and then apply this diff: `arc patch D122814`
* Setup Calypso, and run `yarn start`
* Log out of your account, and log in to [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3Dbfa7ea95b0a07d39e3f23e1d923ce77b35016295d986b51361d4334a6e571206%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1). You will see the "Welcome back!" views have been customized as below:

| Grav | WPJM |
| - | - |
| l_g | l_j |

* Log out of your account. [Disable this submission function](https://github.com/Automattic/wp-calypso/blob/8d6eddd829e42129d7f251b61101174a2a7fc82a/client/login/magic-login/handle-emailed-link-form.jsx#L86), and log in to [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3Dbfa7ea95b0a07d39e3f23e1d923ce77b35016295d986b51361d4334a6e571206%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1). You will see the "Welcome back!" views have been customized as below:

| Grav | WPJM |
| - | - |
| b_g | b_j |

* Log out of your account. Go to your mailbox and then click an expired Gravatar-powered client's login link. The expired link view should still work the same:

截圖 2023-09-25 下午3 10 58

* The WPCOM's [magic login](http://calypso.localhost:3000/log-in/link) still work the same.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by wellyshen** [comment]
@aaronfc Could you mind taking a look at this PR, thanks!

File: client/login/magic-login/emailed-login-link-expired.jsx [hunk 1]
```diff
@@ -17,10 +17,17 @@ class EmailedLoginLinkExpired extends Component {
...
 	};
 
 	componentDidMount() {
 		this.props.recordPageView( '/log-in/link/use', 'Login > Link > Expired' );
+
+		if ( this.props.isGravPoweredClient ) {
+			document
+				.querySelector( '.is-grav-powered-login-page' )
+				?.classList.remove( 'is-grav-powered-login-page' );
+		}
```
Comment (aaronfc) [inline client/login/magic-login/emailed-login-link-expired.jsx:30]: Is there any special reason we want the "link-expired" screen to look like previously? Wouldn't the new one make a better experience?

Comment (aaronfc) [reply client/login/magic-login/emailed-login-link-expired.jsx:30]: I am not fully sure I understand why are we adding/removing these ".is-grav-powered-login-page" like this. Would you mind explaining it a bit?

**Review by aaronfc — APPROVED** [review]
Looks good and works well.

I left a comment because I didn't quite understand what are we doing with th `.is-grav-powered-login-page` and why it is needed to play-around with that class.

File: client/login/magic-login/emailed-login-link-expired.jsx (hunk 1 above)
Comment (wellyshen) [reply client/login/magic-login/emailed-login-link-expired.jsx:30]: > Is there any special reason we want the "link-expired" screen to look like previously? Wouldn't the new one make a better experience? 

I think we will customize the expired link view via this issue: 105859-Automattic/gravatar, by another PR

> I am not fully sure I understand why are we adding/removing these ".is-grav-powered-login-page" like this. Would you mind explaining it a bit?

There's no WPCC's `client_id` URL parameter on the page. So [the way we get the `oauth2Client` data](https://github.com/Automattic/wp-calypso/pull/82049/files#diff-548031d790bc68869d0fb30e0fd46e08effdb1e788d808909a3e776a2104ac04R253-R256) is different from [the `logged-out` component](https://github.com/Automattic/wp-calypso/blob/38ca194f4c2023d800e5fd775cec18edb015a551/client/layout/logged-out.jsx#L258-L261). What we have done here is we get the desired `oauth2Client` and then dynamically hook the `.is-grav-powered-login-page` into the layout element to reuse the existing style at the same file.

**Pull request merged**